charts/consul/values.yaml

```diff
@@ -117,25 +117,44 @@ global:
   # created by this chart. See https://kubernetes.io/docs/concepts/policy/pod-security-policy/.
   enablePodSecurityPolicies: false
 
-  # Configures Consul's gossip encryption key, set as a Kubernetes secret
+  # secretsBackend is used to enable and disable the use of Vault as the secrets backend for the Consul on Kubernetes installation.
+  # In order to utilize this configuration it is required to have an accessible vault cluster which has been pre-configured
+  # by a Vault operator to have the Vault Kube Auth Method, KV2 + PKI secrets engines enabled as well as pre-created
+  # vault secrets, policies and roles.
+  secretsBackend:
+    vault:
+      # Enabling the Vault secrets backend will cause Kubernetes secrets to no longer be used for the following Consul secrets:
+      # - gossip encryption key defined by `global.gossipEncryption.secretName/Key`
```

Comment:
Could you make this description more generic? I think this will a bit be hard to maintain going forward if for every secret we're enabling for vault, we'd also need to update this description